### PR TITLE
Fix: Remove only given package (#4632)

### PR DESCRIPTION
**Pull request by finswimmer**
Fix an issue where all default dependencies were removed instead of just the selected one.

Resolves: https://github.com/python-poetry/poetry/issues/4632

### CODE REVIEW

**Review by abn — DISMISSED** [review]
Looks good, adding some tests might be appropriate here.

File: poetry/core/packages/dependency_group.py [hunk 1]
```diff
@@ -26,13 +26,13 @@ def is_optional(self) -> bool:
...
 
-    def remove_dependency(self, name: "str") -> None:
+    def remove_dependency(self, name: str) -> None:
         from poetry.core.utils.helpers import canonicalize_name
 
         name = canonicalize_name(name)
 
         dependencies = []
-        for dependency in dependencies:
+        for dependency in self.dependencies:
```
[the remark this answers is not in the document]
Comment (finswimmer) [reply poetry/core/packages/dependency_group.py:35]: I've added a test.

File: poetry/core/packages/dependency_group.py [hunk 2]
```diff
@@ -26,13 +26,13 @@ def is_optional(self) -> bool:
     def add_dependency(self, dependency: "DependencyTypes") -> None:
         self._dependencies.append(dependency)
 
-    def remove_dependency(self, name: "str") -> None:
+    def remove_dependency(self, name: str) -> None:
```
[the remark this answers is not in the document]
Comment (finswimmer) [reply poetry/core/packages/dependency_group.py:29]: That change is to small for me to put into a separate commit :smile:

File: poetry/core/packages/dependency_group.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (neersighted) [reply poetry/core/packages/dependency_group.py:35]: Wow, what a bug...

**Review by neersighted — APPROVED** [review]
Looks good! Thanks for adding tests!

**Pull request merged**